exporter/elasticsearchexporter/factory.go

```diff
@@ -37,10 +38,12 @@ func NewFactory() exporter.Factory {
 }
 
 func createDefaultConfig() component.Config {
-	qs := exporterhelper.NewDefaultQueueSettings()
-	qs.Enabled = false
 	return &Config{
-		QueueSettings: qs,
+		QueueSettings: exporterhelper.QueueSettings{
+			Enabled:      false, // FIXME: how does batching without queuing look like?
+			NumConsumers: 100,
```

Comment:
from [QueueSettings](https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/exporterhelper/queue_sender.go#L34)
```
	// NumConsumers is the number of consumers from the queue. Defaults to 10.
	// If batching is enabled, a combined batch cannot contain more requests than the number of consumers.
	// So it's recommended to set higher number of consumers if batching is enabled.
```

Defaulting to 10 means that only 10 pdata structures at max can be in a batch, due to concurrency control in batchsender. e.g. a plog contains 1 log record, that limits it to 10 log records.

The 100 here is still arbitrary, but should be a better default IF sending_queue is enabled, although it is not enabled by default. I'm still thinking if it makes sense to enable sending queue by default.

Reply:
IIANM a request is a `pdata.Logs`, `pdata.Traces`, etc. So not necessarily individual records. Nevertheless, I get your point.

This part of the queue/batch sender API feels kinda awkward to me, and I wonder if it's really desirable. I think this means that if there are 100 (or 10 or whatever) small concurrent requests, then the batch_sender thresholds will never be met. I wonder if the queue consumer should be able to keep consuming until it's had its fill, and then commit, rather than consuming & committing one at a time.

I guess it's fine to default to 100, but maybe open an issue to discuss the API?

Reply:
> This part of the queue/batch sender API feels kinda awkward to me, and I wonder if it's really desirable. I think this means that if there are 100 (or 10 or whatever) small concurrent requests, then the batch_sender thresholds will never be met. I wonder if the queue consumer should be able to keep consuming until it's had its fill, and then commit, rather than consuming & committing one at a time.

I agree that it is awkward to use. I believe that comes from the fact that queue sender spawns a goroutine for each consumer, and that consumer takes care of the lifecycle of the queue entry (from queue to "processing" to deleting). Each of them are doing their thing in parallel. And "num_consumers" control the maximum number of such goroutines. This may require some changes in queue, queue sender and batch sender. I can create an issue to discuss this.

Reply:
https://github.com/open-telemetry/opentelemetry-collector/issues/10368